driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1515,7 +1516,12 @@ ProtocolVersion protocolVersion() {
         }
 
         ThreadFactory threadFactory(String name) {
-            return new ThreadFactoryBuilder().setNameFormat(clusterName + "-" + name + "-%d").build();
+            return new ThreadFactoryBuilder()
```

Comment:
Couldn't we use:

```java
return new DefaultThreadFactory(clusterName + "-" + name);
```

Although I noticed that this gives a slightly different name, i.e.: `cluster1-nio-worker-7-1` instead of `cluster1-nio-worker-1` since it seems to assign a unique pool id to each thread factory, so maybe the way it is done here is best for compatibility with the existing naming (also that unique id seems extraneous since we're sort of doing that ourselves with the cluster name).